sdk/core/azure-core/inc/http/policy.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "context.hpp"
+#include "http.hpp"
+#include "transport.hpp"
+
+namespace Azure { namespace Core { namespace Http {
+
+    class NextHttpPolicy;
+
+  class HttpPolicy {
+  public:
+    // If we get a response that goes up the stack
+    // Any errors in the pipeline throws an exception
+    // At the top of the pipeline we might want to turn certain responses into exceptions
+    virtual std::unique_ptr<Response> Process(
+        Context& context,
+        Request& request,
+        NextHttpPolicy policy) const = 0;
+    virtual ~HttpPolicy() {}
+    //virtual HttpPolicy* Clone() const = 0;
+
+  protected:
+    HttpPolicy() = default;
+    HttpPolicy(const HttpPolicy& other) = default;
+    HttpPolicy(HttpPolicy&& other) = default;
+    HttpPolicy& operator=(const HttpPolicy& other) = default;
+  };
+
+  class NextHttpPolicy {
+    std::size_t m_index;
+    const std::vector<std::unique_ptr<HttpPolicy>>* m_policies;
+
+  public:
+    explicit NextHttpPolicy(const std::vector<std::unique_ptr<HttpPolicy>>* policies)
+        : m_index(0), m_policies(policies)
+    {
+    }
+
+    explicit NextHttpPolicy(
+        std::size_t index,
+        const std::vector<std::unique_ptr<HttpPolicy>>* policies)
+        : m_index(index), m_policies(policies)
+    {
+    }
+
+    std::unique_ptr<Response> ProcessNext(Context& ctx, Request& req);
```

Comment:
I'd just call this Send too. The type is call NextPolcy (no Http if the namespace is enough of a disambiguator) and then I call Send on the "Next Policy"; no need for the method to be named SendNext (or ProcessNext).

Reply:
Good catch that method isn't used anymore.

Reply:
Send will work here, its consistent and only us writing the code might get confused on which send is being called